include/CL/cl_exp_defined_builtin_kernels.h

```diff
@@ -209,4 +210,23 @@ typedef struct
   cl_int quality;
 } cl_dbk_attributes_exp_jpeg_encode;
 
+/* Name: "exp_onnx_inference"
+ * Attributes for constructing an inference session for ONNX format ML models.
+ */
+typedef struct _cl_dbk_attributes_exp_onnx_inference
+{
+  char *model_data;
```

Comment:
All pointer, in the DBK attribute structures, should be `const` qualified. 
```suggestion
  const char *model_data;
```

Reply:
That would make things more cumbersome to use, I've opted instead to make the instances of the attributes const. Maybe in the future the program->builtin_kernel_attributes could be made const.

Reply:
We should be able to treat all attribute fields as immutable at all times I think? The only exception would be of course creating and deleting the struct. But that can be tucked away in helper functions that use separate local variables to assemble the fields and only write the final data in the struct when they have everything ready. And for zeroing out the struct upon deletion we can sidestep constness entirely with memset.

Reply:
yes they should be fine treated as immutable, the original attributes passed over to `clCreateFunctionWithDefinedBuiltinKernels` are deep copied. It's just annoying for an end user to make sure all the strings and data is const (cast) when creating the attribute instance.

Reply:
I'm doubtful that Khronos will accept structures with non-`const` pointers in them. See `cl_mutable_*` structures in the OpenCL API spec - all of the pointer members are `const` qualified. 

However, there is at least one exception to this pattern: `cl_dx9_surface_info_khr` structure.

Reply:
I see, what should be done with the double pointer cases? Should there be a set limit and use a const array, or something along the lines of `const char * const *` ?

Reply:
I think `const TYPE **` form is enough. I don't see any `const TYPE * const *`s in the core API spec.

Reply:
also, I noticed that `cl_tensor_desc.layout` is of type `void *` shouldn't that also be a `const` then?

Reply:
Yeah, that probably should be changed too. I'll do that when I get time to.

Reply:
oke the code has been changed to use const for pointers. I also found out that clReleaseProgram didn't free kernel attributes, so i've made a separate commit for that.

Reply:
@linehill for review, besides the struct, this is the file that has been changed with the `const` changes: https://github.com/pocl/pocl/pull/1567/files#diff-cebcbd0d759fb1fd119d11af3678805dc635025332ee1c2e0860a3fad5a99c28